EWS Listener throws AddressException when from address contains whitespaces.

**Describe the bug**
EWS Listener throws AddressException when from address contains whitespaces.

**Reporter**
Laurens Mäkel

**Category** 
Use following Tags to categorize your bug report
 - Frank Framework

**To Reproduce**
I forwarded the e-mail to Gerrit.

**Expected behavior**
E-mail to be processed by pipeline.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment:**
 - Application server (f.e. Tomcat/WebSphere/JBOSS/Jetty) Websphere
 - Browser [e.g. chrome, safari] Chrome 
 - IAF Version [e.g. 7.5 RC3, 7.6]  7.5-20191008.103430

**Additional context**
2020-11-19 14:57:26,056 ERROR [HandleMail_pensioenuitkeringen@nn.nl-listener[273]] cid [AAMkADk5OWQyYjJmLTc1NzMtNGYyNy05ZjA1LWUxM2NkMDZmYTQ3NQBGAAAAAAA8HnzbPlI+TYBpWG1N7naLBwDCz7h0HvtjRIdfhNCSS1LuAAAAADdAAADCz7h0HvtjRIdfhNCSS1LuAAAeOB2WAAA=] core.Adapter - Adapter [HandleMail_pensioenuitkeringen@nn.nl] error processing message with messageId [AAMkADk5OWQyYjJmLTc1NzMtNGYyNy05ZjA1LWUxM2NkMDZmYTQ3NQBGAAAAAAA8HnzbPlI+TYBpWG1N7naLBwDCz7h0HvtjRIdfhNCSS1LuAAAAADdAAADCz7h0HvtjRIdfhNCSS1LuAAAeOB2WAAA=]:
nl.nn.adapterframework.core.ListenerException: Pipe [Send_mail] msgId [AAMkADk5OWQyYjJmLTc1NzMtNGYyNy05ZjA1LWUxM2NkMDZmYTQ3NQBGAAAAAAA8HnzbPlI+TYBpWG1N7naLBwDCz7h0HvtjRIdfhNCSS1LuAAAAADdAAADCz7h0HvtjRIdfhNCSS1LuAAAeOB2WAAA=] caught exception: MailSender [Send_mail-sender] cannot connect send message to smtpHost [mail.insim.biz]: (AddressException) [popp.dk scan@popp.dk]: Local address contains control or whitespace
      at nl.nn.adapterframework.core.Adapter.processMessageWithExceptions(Adapter.java:675)
      at nl.nn.adapterframework.receivers.ReceiverBase.processMessageInAdapter(ReceiverBase.java:1168)
      at nl.nn.adapterframework.receivers.ReceiverBase.processRawMessage(ReceiverBase.java:972)
      at nl.nn.adapterframework.receivers.ReceiverBase.processRawMessage(ReceiverBase.java:951)
      at nl.nn.adapterframework.receivers.ReceiverBase.processRawMessage(ReceiverBase.java:946)
      at nl.nn.adapterframework.receivers.PullingListenerContainer$ListenTask.run(PullingListenerContainer.java:206)
      at java.lang.Thread.run(Thread.java:818)
Caused by: nl.nn.adapterframework.core.PipeRunException: Pipe [Send_mail] msgId [AAMkADk5OWQyYjJmLTc1NzMtNGYyNy05ZjA1LWUxM2NkMDZmYTQ3NQBGAAAAAAA8HnzbPlI+TYBpWG1N7naLBwDCz7h0HvtjRIdfhNCSS1LuAAAAADdAAADCz7h0HvtjRIdfhNCSS1LuAAAeOB2WAAA=] caught exception: MailSender [Send_mail-sender] cannot connect send message to smtpHost [mail.insim.biz]: (AddressException) [popp.dk scan@popp.dk]: Local address contains control or whitespace
      at nl.nn.adapterframework.pipes.MessageSendingPipe.doPipe(MessageSendingPipe.java:727)
      at nl.nn.adapterframework.stream.StreamingPipe.doPipe(StreamingPipe.java:95)
      at nl.nn.adapterframework.processors.CorePipeProcessor.processPipe(CorePipeProcessor.java:30)
      at nl.nn.adapterframework.processors.CheckMessageSizePipeProcessor.processPipe(CheckMessageSizePipeProcessor.java:37)
      at nl.nn.adapterframework.processors.LockerPipeProcessor.processPipe(LockerPipeProcessor.java:58)
      at nl.nn.adapterframework.processors.TransactionAttributePipeProcessor.processPipe(TransactionAttributePipeProcessor.java:61)
      at nl.nn.adapterframework.processors.CheckSemaphorePipeProcessor.processPipe(CheckSemaphorePipeProcessor.java:57)
      at nl.nn.adapterframework.processors.CheckSemaphorePipeProcessor$$FastClassBySpringCGLIB$$3159a29c.invoke(<generated>)
      at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
      at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:700)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
      at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:91)
      at nl.nn.ibistesttool.IbisDebuggerAdvice.debugPipeGetInputFrom(IbisDebuggerAdvice.java:116)
      at sun.reflect.GeneratedMethodAccessor342.invoke(Unknown Source)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
      at java.lang.reflect.Method.invoke(Method.java:508)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
      at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:65)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
      at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
      at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:633)
      at nl.nn.adapterframework.processors.CheckSemaphorePipeProcessor$$EnhancerBySpringCGLIB$$86dd1b47.processPipe(<generated>)
      at nl.nn.adapterframework.processors.InputOutputPipeProcessor.processPipe(InputOutputPipeProcessor.java:78)
      at nl.nn.adapterframework.processors.InputOutputPipeProcessor$$FastClassBySpringCGLIB$$feb9be85.invoke(<generated>)
      at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
      at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:700)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
      at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:91)
      at nl.nn.ibistesttool.IbisDebuggerAdvice.debugPipeInputOutputAbort(IbisDebuggerAdvice.java:89)
      at sun.reflect.GeneratedMethodAccessor341.invoke(Unknown Source)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
      at java.lang.reflect.Method.invoke(Method.java:508)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
      at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:65)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
      at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
      at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:633)
      at nl.nn.adapterframework.processors.InputOutputPipeProcessor$$EnhancerBySpringCGLIB$$de90cb10.processPipe(<generated>)
      at nl.nn.adapterframework.processors.MonitoringPipeProcessor.processPipe(MonitoringPipeProcessor.java:78)
      at nl.nn.adapterframework.processors.CorePipeLineProcessor.processPipeLine(CorePipeLineProcessor.java:162)
      at nl.nn.adapterframework.processors.CorePipeLineProcessor$$FastClassBySpringCGLIB$$cf1091e5.invoke(<generated>)
      at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
      at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:700)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
      at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:91)
      at nl.nn.ibistesttool.IbisDebuggerAdvice.debugPipeLineInputOutputAbort(IbisDebuggerAdvice.java:71)
      at sun.reflect.GeneratedMethodAccessor517.invoke(Unknown Source)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
      at java.lang.reflect.Method.invoke(Method.java:508)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
      at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:65)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
      at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
      at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:633)
      at nl.nn.adapterframework.processors.CorePipeLineProcessor$$EnhancerBySpringCGLIB$$c46f9270.processPipeLine(<generated>)
      at nl.nn.adapterframework.processors.LockerPipeLineProcessor.processPipeLine(LockerPipeLineProcessor.java:68)
      at nl.nn.adapterframework.processors.TransactionAttributePipeLineProcessor.processPipeLine(TransactionAttributePipeLineProcessor.java:49)
      at nl.nn.adapterframework.processors.CachePipeLineProcessor.processPipeLine(CachePipeLineProcessor.java:35)
      at nl.nn.adapterframework.processors.InputOutputPipeLineProcessor.processPipeLine(InputOutputPipeLineProcessor.java:48)
      at nl.nn.adapterframework.core.PipeLine.process(PipeLine.java:539)
      at nl.nn.adapterframework.core.Adapter.processMessageWithExceptions(Adapter.java:655)
      ... 6 more
Caused by: nl.nn.adapterframework.core.SenderException: MailSender [Send_mail-sender] cannot connect send message to smtpHost [mail.insim.biz]: (AddressException) [popp.dk scan@popp.dk]: Local address contains control or whitespace
      at nl.nn.adapterframework.senders.MailSender.putOnTransport(MailSender.java:436)
      at nl.nn.adapterframework.senders.MailSender.sendEmail(MailSender.java:298)
      at nl.nn.adapterframework.senders.MailSender.sendEmail(MailSender.java:210)
      at nl.nn.adapterframework.senders.MailSenderBase.sendMessage(MailSenderBase.java:74)
      at nl.nn.adapterframework.senders.MailSenderBase$$FastClassBySpringCGLIB$$f65be02e.invoke(<generated>)
      at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
      at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:700)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
      at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:91)
      at nl.nn.ibistesttool.IbisDebuggerAdvice.debugSenderInputAbort(IbisDebuggerAdvice.java:251)
      at nl.nn.ibistesttool.IbisDebuggerAdvice.debugSenderWithParametersInputOutputAbort(IbisDebuggerAdvice.java:140)
      at sun.reflect.GeneratedMethodAccessor514.invoke(Unknown Source)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
      at java.lang.reflect.Method.invoke(Method.java:508)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
      at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
      at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:65)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
      at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
      at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
      at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:633)
      at nl.nn.adapterframework.senders.MailSender$$EnhancerBySpringCGLIB$$6db39a5.sendMessage(<generated>)
      at nl.nn.adapterframework.pipes.MessageSendingPipe.sendTextMessage(MessageSendingPipe.java:856)
      at nl.nn.adapterframework.pipes.MessageSendingPipe.sendMessage(MessageSendingPipe.java:802)
      at nl.nn.adapterframework.pipes.MessageSendingPipe.doPipe(MessageSendingPipe.java:543)
      ... 72 more
Caused by: javax.mail.internet.AddressException: Local address contains control or whitespace in string ``popp.dk scan@popp.dk''
      at javax.mail.internet.InternetAddress.checkAddress(InternetAddress.java:1213)
      at javax.mail.internet.InternetAddress.parse(InternetAddress.java:1091)
      at javax.mail.internet.InternetAddress.parseHeader(InternetAddress.java:658)
      at javax.mail.internet.MimeMessage.getAddressHeader(MimeMessage.java:702)
      at javax.mail.internet.MimeMessage.getFrom(MimeMessage.java:362)
      at com.sun.mail.smtp.SMTPTransport.mailFrom(SMTPTransport.java:1555)
      at com.sun.mail.smtp.SMTPTransport.sendMessage(SMTPTransport.java:1119)
      at nl.nn.adapterframework.senders.MailSender.putOnTransport(MailSender.java:432)
      ... 96 more

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
@@ -698,6 +698,38 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 	}
 
 	@Override
+	public String getSender(EmailMessage emailMessage) throws FileSystemException {
+		try {
+			EmailAddress sender = emailMessage.getSender();
+			return sender==null ? null : sender.getAddress();
+		} catch (ServiceLocalException e) {
+			log.warn("Could not get Sender Address: "+ e.getMessage());
+			return null;
+		}
+	}
+
+	@Override
+	public String getReplyTo(EmailMessage emailMessage) throws FileSystemException {
+		try {
+			EmailAddressCollection replyTo = emailMessage.getReplyTo();
+			if (replyTo==null) {
+				return null;
+			}
+			
+			String addressList=null;
+			for (EmailAddress address: replyTo) {
+				if (addressList!=null) {
+					addressList += ", ";
+				}
+				addressList += address.getAddress();
+			}
+			return MailFileSystemUtils.getValidAddress("replyTo", addressList);
+		} catch (ServiceLocalException e) {
+			throw new FileSystemException("Could not get ReplyTo Address", e);
+		}
+	}
+	
+	@Override
 	public String getSubject(EmailMessage emailMessage) throws FileSystemException {
 		try {
 			return emailMessage.getSubject();

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
@@ -114,6 +114,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 	private String filter;
 	private boolean readMimeContents=false;
 	private int maxNumberOfMessagesToList=10;
+	private String replyAddressFields = REPLY_ADDRESS_FIELDS_DEFAULT;
 
 	private String proxyHost = null;
 	private int proxyPort = 8080;
@@ -484,17 +485,17 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 					result.put(internetMessageHeader.getName(), internetMessageHeader.getValue());
 					continue;
 				}
-				List values;
+				List<Object> values;
 				if (curEntry instanceof List) {
-					values = (List)curEntry;
+					values = (List<Object>)curEntry;
 				} else {
-					values = new LinkedList();
+					values = new LinkedList<Object>();
 					values.add(curEntry);
 					result.put(internetMessageHeader.getName(),values);
 				}
 				values.add(internetMessageHeader.getValue());
 			}
-			
+			result.put(IMailFileSystem.BEST_REPLY_ADDRESS_KEY, MailFileSystemUtils.findBestReplyAddress(result,getReplyAddressFields()));
 			return result;
 		} catch (Exception e) {
 			throw new FileSystemException(e);
@@ -680,7 +681,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		}
 	}
 
-	public List getReplyTo(EmailMessage emailMessage) throws FileSystemException {
+	public List<String> getReplyTo(EmailMessage emailMessage) throws FileSystemException {
 		try {
 			EmailAddressCollection replyTo = emailMessage.getReplyTo();
 			if (replyTo==null) {
@@ -844,22 +845,32 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return filter;
 	}
 
-	public boolean isReadMimeContents() {
-		return readMimeContents;
+	@IbisDoc({"10", "Comma separated list of fields to try as response address", REPLY_ADDRESS_FIELDS_DEFAULT})
+	public void setReplyAddressFields(String replyAddressFields) {
+		this.replyAddressFields = replyAddressFields;
 	}
+	@Override
+	public String getReplyAddressFields() {
+		return replyAddressFields;
+	}
+
+	@IbisDoc({"11", "if set <code>true</code>, the contents will be read in MIME format", "false"})
 	public void setReadMimeContents(boolean readMimeContents) {
 		this.readMimeContents = readMimeContents;
 	}
-
-
-	public int getMaxNumberOfMessagesToList() {
-		return maxNumberOfMessagesToList;
+	public boolean isReadMimeContents() {
+		return readMimeContents;
 	}
+
+	@IbisDoc({"12", "the maximum number of messages to be retrieved from a folder", "10"})
 	public void setMaxNumberOfMessagesToList(int maxNumberOfMessagesToList) {
 		this.maxNumberOfMessagesToList = maxNumberOfMessagesToList;
 	}
+	public int getMaxNumberOfMessagesToList() {
+		return maxNumberOfMessagesToList;
+	}
 
-	@IbisDoc({"10", "proxy host", ""})
+	@IbisDoc({"13", "proxy host", ""})
 	public void setProxyHost(String proxyHost) {
 		this.proxyHost = proxyHost;
 	}
@@ -867,7 +878,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return proxyHost;
 	}
 
-	@IbisDoc({"11", "proxy port", ""})
+	@IbisDoc({"14", "proxy port", ""})
 	public void setProxyPort(int proxyPort) {
 		this.proxyPort = proxyPort;
 	}
@@ -875,7 +886,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return proxyPort;
 	}
 
-	@IbisDoc({"12", "proxy username", ""})
+	@IbisDoc({"15", "proxy username", ""})
 	public void setProxyUsername(String proxyUsername) {
 		this.proxyUsername = proxyUsername;
 	}
@@ -883,7 +894,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return proxyUsername;
 	}
 
-	@IbisDoc({"13", "proxy password", ""})
+	@IbisDoc({"16", "proxy password", ""})
 	public void setProxyPassword(String proxyPassword) {
 		this.proxyPassword = proxyPassword;
 	}
@@ -891,7 +902,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return proxyPassword;
 	}
 
-	@IbisDoc({"14", "proxy authAlias", ""})
+	@IbisDoc({"17", "proxy authAlias", ""})
 	public void setProxyAuthAlias(String proxyAuthAlias) {
 		this.proxyAuthAlias = proxyAuthAlias;
 	}
@@ -899,7 +910,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		return proxyAuthAlias;
 	}
 
-	@IbisDoc({"15", "proxy domain", ""})
+	@IbisDoc({"18", "proxy domain", ""})
 	public void setProxyDomain(String proxyDomain) {
 		this.proxyDomain = proxyDomain;
 	}

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java
@@ -470,6 +470,14 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 				emailMessage = f;
 			}
 			Map<String, Object> result=new LinkedHashMap<String,Object>();
+			result.put(IMailFileSystem.TO_RECEPIENTS_KEY, asList(emailMessage.getToRecipients()));
+			result.put(IMailFileSystem.CC_RECEPIENTS_KEY, asList(emailMessage.getCcRecipients()));
+			result.put(IMailFileSystem.BCC_RECEPIENTS_KEY, asList(emailMessage.getBccRecipients()));
+			result.put(IMailFileSystem.FROM_ADDRESS_KEY, emailMessage.getFrom().getAddress()); 
+			result.put(IMailFileSystem.SENDER_ADDRESS_KEY, getSender(emailMessage)); 
+			result.put(IMailFileSystem.REPLY_TO_RECEPIENTS_KEY, getReplyTo(emailMessage)); 
+			result.put(IMailFileSystem.DATETIME_SENT_KEY, emailMessage.getDateTimeSent()); 
+			result.put(IMailFileSystem.DATETIME_RECEIVED_KEY, emailMessage.getDateTimeReceived()); 
 			for(InternetMessageHeader internetMessageHeader : emailMessage.getInternetMessageHeaders()) {
 				Object curEntry = result.get(internetMessageHeader.getName());
 				if (curEntry==null) {
@@ -661,43 +669,7 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		}
 	}
 
-	@Override
-	public Iterable<String> getToRecipients(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return asList(emailMessage.getToRecipients());
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get ToRecipients", e);
-		}
-	}
-
-	@Override
-	public Iterable<String> getCCRecipients(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return asList(emailMessage.getCcRecipients());
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get CCRecipients", e);
-		}
-	}
 	
-	@Override
-	public Iterable<String> getBCCRecipients(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return asList(emailMessage.getBccRecipients());
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get BCCRecipients", e);
-		}
-	}
-	
-	@Override
-	public String getFrom(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return emailMessage.getFrom().getAddress();
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get From Address", e);
-		}
-	}
-
-	@Override
 	public String getSender(EmailMessage emailMessage) throws FileSystemException {
 		try {
 			EmailAddress sender = emailMessage.getSender();
@@ -708,24 +680,19 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		}
 	}
 
-	@Override
-	public String getReplyTo(EmailMessage emailMessage) throws FileSystemException {
+	public List getReplyTo(EmailMessage emailMessage) throws FileSystemException {
 		try {
 			EmailAddressCollection replyTo = emailMessage.getReplyTo();
 			if (replyTo==null) {
 				return null;
 			}
-			
-			String addressList=null;
+			List<String> result = new LinkedList<>();
 			for (EmailAddress address: replyTo) {
-				if (addressList!=null) {
-					addressList += ", ";
-				}
-				addressList += address.getAddress();
+				result.add(address.getAddress());
 			}
-			return MailFileSystemUtils.getValidAddress("replyTo", addressList);
+			return result;
 		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get ReplyTo Address", e);
+			throw new FileSystemException("Could not get ReplyTo Addresses", e);
 		}
 	}
 	
@@ -738,23 +705,6 @@ public class ExchangeFileSystem implements IMailFileSystem<EmailMessage,Attachme
 		}
 	}
 	
-	@Override
-	public Date getDateTimeSent(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return emailMessage.getDateTimeSent();
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get DateTimeSent", e);
-		}
-	}
-
-	@Override
-	public Date getDateTimeReceived(EmailMessage emailMessage) throws FileSystemException {
-		try {
-			return emailMessage.getDateTimeReceived();
-		} catch (ServiceLocalException e) {
-			throw new FileSystemException("Could not get DateTimeReceived", e);
-		}
-	}
 
 	@Override
 	public String getMessageBody(EmailMessage emailMessage) throws FileSystemException {

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
@@ -25,25 +25,21 @@ public interface IMailFileSystem<M,A> extends IWithAttachments<M,A> {
 	public String MAIL_MESSAGE_ID = "Message-ID";
 	
 	public Iterable<String> getToRecipients(M emailMessage) throws FileSystemException ;
-
 	public Iterable<String> getCCRecipients(M emailMessage) throws FileSystemException;
-	
 	public Iterable<String> getBCCRecipients(M emailMessage) throws FileSystemException;
 	
 	public String getFrom(M emailMessage) throws FileSystemException;
+	public String getSender(M emailMessage) throws FileSystemException;
+	public String getReplyTo(M emailMessage) throws FileSystemException;
 
 	public String getSubject(M emailMessage) throws FileSystemException;
-	
 	public Date getDateTimeSent(M emailMessage) throws FileSystemException;
-	
 	public Date getDateTimeReceived(M emailMessage) throws FileSystemException;
 
 	public String getMessageBody(M emailMessage) throws FileSystemException;
-	
 	public Message getMimeContent(M emailMessage) throws FileSystemException;
 
 	public void extractEmail(M emailMessage, SaxElementBuilder emailXml) throws FileSystemException;
-	
 	public void extractAttachment(A attachment, SaxElementBuilder attachmentsXml) throws FileSystemException;
 	
 }

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
@@ -15,26 +15,28 @@
 */
 package nl.nn.adapterframework.filesystem;
 
-import java.util.Date;
-
 import nl.nn.adapterframework.stream.Message;
 import nl.nn.adapterframework.xml.SaxElementBuilder;
 
 public interface IMailFileSystem<M,A> extends IWithAttachments<M,A> {
 
-	public String MAIL_MESSAGE_ID = "Message-ID";
+	public final String MAIL_MESSAGE_ID = "Message-ID";
+	public final String RETURN_PATH_HEADER="Return-Path";
 	
-	public Iterable<String> getToRecipients(M emailMessage) throws FileSystemException ;
-	public Iterable<String> getCCRecipients(M emailMessage) throws FileSystemException;
-	public Iterable<String> getBCCRecipients(M emailMessage) throws FileSystemException;
-	
-	public String getFrom(M emailMessage) throws FileSystemException;
-	public String getSender(M emailMessage) throws FileSystemException;
-	public String getReplyTo(M emailMessage) throws FileSystemException;
+	/*
+	 * IMailFileSystems should make sure the additionalProperties contain also the following keys:
+	 */
+	public final String TO_RECEPIENTS_KEY = "to";
+	public final String CC_RECEPIENTS_KEY = "cc";
+	public final String BCC_RECEPIENTS_KEY = "bcc";
+	public final String FROM_ADDRESS_KEY = "from";     // originator of the message
+	public final String SENDER_ADDRESS_KEY = "sender"; // identifies who submitted the messages, probably on behalf of the 'From'
+	public final String REPLY_TO_RECEPIENTS_KEY = "replyTo";
+	public final String DATETIME_SENT_KEY = "DateTimeSent";         // as Date, or in XML format: yyyy-MM-dd'T'HH:mm:ss.SSSZ
+	public final String DATETIME_RECEIVED_KEY = "DateTimeReceived";
 
+	
 	public String getSubject(M emailMessage) throws FileSystemException;
-	public Date getDateTimeSent(M emailMessage) throws FileSystemException;
-	public Date getDateTimeReceived(M emailMessage) throws FileSystemException;
 
 	public String getMessageBody(M emailMessage) throws FileSystemException;
 	public Message getMimeContent(M emailMessage) throws FileSystemException;

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/IMailFileSystem.java
@@ -35,6 +35,9 @@ public interface IMailFileSystem<M,A> extends IWithAttachments<M,A> {
 	public final String DATETIME_SENT_KEY = "DateTimeSent";         // as Date, or in XML format: yyyy-MM-dd'T'HH:mm:ss.SSSZ
 	public final String DATETIME_RECEIVED_KEY = "DateTimeReceived";
 
+	public final String BEST_REPLY_ADDRESS_KEY = "bestReplyAddress";
+	public final String REPLY_ADDRESS_FIELDS_DEFAULT=REPLY_TO_RECEPIENTS_KEY+','+FROM_ADDRESS_KEY+','+SENDER_ADDRESS_KEY+','+RETURN_PATH_HEADER;
+
 	
 	public String getSubject(M emailMessage) throws FileSystemException;
 
@@ -43,5 +46,7 @@ public interface IMailFileSystem<M,A> extends IWithAttachments<M,A> {
 
 	public void extractEmail(M emailMessage, SaxElementBuilder emailXml) throws FileSystemException;
 	public void extractAttachment(A attachment, SaxElementBuilder attachmentsXml) throws FileSystemException;
+	
+	public String getReplyAddressFields();
 	
 }

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
@@ -44,7 +44,8 @@ public class MailFileSystemUtils {
 				IMailFileSystem.SENDER_ADDRESS_KEY,
 				IMailFileSystem.REPLY_TO_RECEPIENTS_KEY,
 				IMailFileSystem.DATETIME_SENT_KEY,
-				IMailFileSystem.DATETIME_RECEIVED_KEY } );
+				IMailFileSystem.DATETIME_RECEIVED_KEY,
+				IMailFileSystem.BEST_REPLY_ADDRESS_KEY} );
 
 	
 	public static <M,A> void addEmailInfoSimple(IMailFileSystem<M,A> fileSystem, M emailMessage, SaxElementBuilder emailXml) throws FileSystemException, SAXException {
@@ -92,8 +93,7 @@ public class MailFileSystemUtils {
 		addPropertyAsHeader(emailXml,IMailFileSystem.FROM_ADDRESS_KEY, properties.get(IMailFileSystem.FROM_ADDRESS_KEY));
 		addPropertyAsHeader(emailXml,IMailFileSystem.SENDER_ADDRESS_KEY, properties.get(IMailFileSystem.SENDER_ADDRESS_KEY));
 		addPropertyAsHeader(emailXml,IMailFileSystem.REPLY_TO_RECEPIENTS_KEY, properties.get(IMailFileSystem.REPLY_TO_RECEPIENTS_KEY));
-		String bestReplyAddress = findBestReplyAddress(properties);
-		emailXml.addElement("bestReplyAddress", bestReplyAddress);
+		addPropertyAsHeader(emailXml,IMailFileSystem.BEST_REPLY_ADDRESS_KEY, properties.get(IMailFileSystem.BEST_REPLY_ADDRESS_KEY));
 		emailXml.addElement("subject", fileSystem.getSubject(emailMessage));
 		addPropertyAsHeader(emailXml,IMailFileSystem.DATETIME_SENT_KEY, properties.get(IMailFileSystem.DATETIME_SENT_KEY));
 		addPropertyAsHeader(emailXml,IMailFileSystem.DATETIME_RECEIVED_KEY, properties.get(IMailFileSystem.DATETIME_RECEIVED_KEY));
@@ -131,20 +131,15 @@ public class MailFileSystemUtils {
 	}
 	
 
-	public static String findBestReplyAddress(Map<String,Object> headers) {
+	public static String findBestReplyAddress(Map<String,Object> headers, String replyAddressFields) {
+		if (StringUtils.isEmpty(replyAddressFields)) {
+			return null;
+		}
 		String result;
-		
-		if (null != (result = getValidAddressFromHeader(IMailFileSystem.REPLY_TO_RECEPIENTS_KEY, headers))) {
-			return result; 
-		}
-		if (null != (result = getValidAddressFromHeader(IMailFileSystem.FROM_ADDRESS_KEY, headers))) {
-			return result; 
-		}
-		if (null != (result = getValidAddressFromHeader(IMailFileSystem.SENDER_ADDRESS_KEY, headers))) {
-			return result; 
-		}
-		if (null != (result = getValidAddressFromHeader(IMailFileSystem.RETURN_PATH_HEADER, headers))) {
-			return result; 
+		for(String field:replyAddressFields.split(",")) {
+			if (null != (result = getValidAddressFromHeader(field, headers))) {
+				return result; 
+			}
 		}
 		return null;
 	}

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
@@ -20,6 +20,10 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.xml.sax.SAXException;
 
@@ -28,6 +32,8 @@ import nl.nn.adapterframework.xml.SaxElementBuilder;
 
 public class MailFileSystemUtils {
 	protected static Logger log = LogUtil.getLogger(MailFileSystemUtils.class);
+	
+	public static final String RETURN_PATH_HEADER="Return-Path";
 
 	public static <M,A> void addEmailInfoSimple(IMailFileSystem<M,A> fileSystem, M emailMessage, SaxElementBuilder emailXml) throws FileSystemException, SAXException {
 		emailXml.addElement("subject", fileSystem.getSubject(emailMessage));
@@ -45,7 +51,15 @@ public class MailFileSystemUtils {
 				recipientsXml.addElement("recipient", "type", "bcc", address);
 			}
 		}
-		emailXml.addElement("from", fileSystem.getFrom(emailMessage));
+		Map<String,Object> properties = fileSystem.getAdditionalFileProperties(emailMessage);
+		String from    = fileSystem.getFrom(emailMessage);
+		String replyTo = fileSystem.getReplyTo(emailMessage);
+		String sender  = fileSystem.getSender(emailMessage);
+		String bestReplyAddress = findBestReplyAddress(from, replyTo, sender, properties);
+		if (StringUtils.isNotEmpty(from))    emailXml.addElement("from", from);
+		if (StringUtils.isNotEmpty(replyTo)) emailXml.addElement("replyTo", replyTo);
+		if (StringUtils.isNotEmpty(sender)) emailXml.addElement("sender", sender);
+		emailXml.addElement("bestReplyAddress", bestReplyAddress);
 		emailXml.addElement("subject", fileSystem.getSubject(emailMessage));
 		emailXml.addElement("message", fileSystem.getMessageBody(emailMessage));
 		try (SaxElementBuilder attachmentsXml = emailXml.startElement("attachments")) {
@@ -56,7 +70,6 @@ public class MailFileSystemUtils {
 			throw new FileSystemException("Cannot extract attachment",e);
 		}
 		try (SaxElementBuilder headersXml = emailXml.startElement("headers")) {
-			Map<String,Object> properties = fileSystem.getAdditionalFileProperties(emailMessage);
 			if (properties != null) {
 				for (Map.Entry<String,Object> header: properties.entrySet()) {
 					if (header.getValue() instanceof List) {
@@ -85,6 +98,42 @@ public class MailFileSystemUtils {
 			if (emailMessage!=null) {
 				fileSystem.extractEmail(emailMessage, attachmentXml);
 			}
+		}
+	}
+	
+
+	public static String findBestReplyAddress(String from, String replyTo, String sender, Map<String,Object> headers) {
+		String result;
+		
+		if (null != (result = getValidAddress("replyTo", replyTo))) {
+			return result; 
+		}
+		if (null != (result = getValidAddress("from", from))) {
+			return result; 
+		}
+		if (null != (result = getValidAddress("sender", sender))) {
+			return result; 
+		}
+		Object returnPath = headers.get(RETURN_PATH_HEADER);
+		if (returnPath instanceof String && null != (result = getValidAddress("return path", (String)returnPath))) {
+			return result; 
+		}
+		return null;
+	}
+	
+	public static String getValidAddress(String type, String address) {
+		try {
+			if (StringUtils.isEmpty(address)) {
+				return null;
+			}
+			InternetAddress[] addresses = InternetAddress.parseHeader(address, true);
+			if (addresses.length==0) {
+				return null;
+			} 
+			return InternetAddress.toString(addresses);
+		} catch (AddressException e) {
+			log.warn("type ["+type+"] address ["+address+"] is invalid: "+e.getMessage());
+			return null;
 		}
 	}
 	

--- a/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
+++ b/core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java
@@ -16,6 +16,8 @@
 package nl.nn.adapterframework.filesystem;
 
 import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
@@ -33,34 +35,68 @@ import nl.nn.adapterframework.xml.SaxElementBuilder;
 public class MailFileSystemUtils {
 	protected static Logger log = LogUtil.getLogger(MailFileSystemUtils.class);
 	
-	public static final String RETURN_PATH_HEADER="Return-Path";
+	@SuppressWarnings("unchecked")
+	public static final List<String> specialHeaders = Arrays.asList(new String[] {
+				IMailFileSystem.TO_RECEPIENTS_KEY, 
+				IMailFileSystem.CC_RECEPIENTS_KEY,
+				IMailFileSystem.BCC_RECEPIENTS_KEY,
+				IMailFileSystem.FROM_ADDRESS_KEY,
+				IMailFileSystem.SENDER_ADDRESS_KEY,
+				IMailFileSystem.REPLY_TO_RECEPIENTS_KEY,
+				IMailFileSystem.DATETIME_SENT_KEY,
+				IMailFileSystem.DATETIME_RECEIVED_KEY } );
 
+	
 	public static <M,A> void addEmailInfoSimple(IMailFileSystem<M,A> fileSystem, M emailMessage, SaxElementBuilder emailXml) throws FileSystemException, SAXException {
 		emailXml.addElement("subject", fileSystem.getSubject(emailMessage));
 	}
 
-	public static <M,A> void addEmailInfo(IMailFileSystem<M,A> fileSystem, M emailMessage, SaxElementBuilder emailXml) throws FileSystemException, SAXException {
-		try (SaxElementBuilder recipientsXml = emailXml.startElement("recipients")) {
-			for(String address:fileSystem.getToRecipients(emailMessage)) {
-				recipientsXml.addElement("recipient", "type", "to", address);
-			}
-			for(String address:fileSystem.getCCRecipients(emailMessage)) {
-				recipientsXml.addElement("recipient", "type", "cc", address);
-			}
-			for(String address:fileSystem.getBCCRecipients(emailMessage)) {
-				recipientsXml.addElement("recipient", "type", "bcc", address);
-			}
+	public static void addPropertyAsHeader(SaxElementBuilder xml, String elementName, Object property) throws SAXException {
+		addPropertyAsHeader(xml, elementName, null, null, property);
+	}
+	
+	public static void addPropertyAsHeader(SaxElementBuilder xml, String elementName, String attributeName, String attributeValue, Object property) throws SAXException {
+		if (property==null) {
+			return;
 		}
+		if (property instanceof Iterable) {
+			for (Object item:(Iterable)property) {
+				addPropertyAsHeader(xml, elementName, attributeName, attributeValue, item);
+			}
+			return;
+ 		}
+		String value;
+		if (property instanceof Date) {
+			value = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format((Date)property);
+		} else {
+			value = property.toString();
+		}
+		if (attributeName!=null) {
+			xml.addElement(elementName, attributeName, attributeValue, value);
+		} else {
+			xml.addElement(elementName, value);
+		}
+	}
+	
+	public static void addRecipientProperty(SaxElementBuilder recipientsXml, Map<String,Object> properties, String key) throws SAXException {
+		addPropertyAsHeader(recipientsXml,"recipient", "type", key, properties.get(key));
+	}
+	
+	public static <M,A> void addEmailInfo(IMailFileSystem<M,A> fileSystem, M emailMessage, SaxElementBuilder emailXml) throws FileSystemException, SAXException {
 		Map<String,Object> properties = fileSystem.getAdditionalFileProperties(emailMessage);
-		String from    = fileSystem.getFrom(emailMessage);
-		String replyTo = fileSystem.getReplyTo(emailMessage);
-		String sender  = fileSystem.getSender(emailMessage);
-		String bestReplyAddress = findBestReplyAddress(from, replyTo, sender, properties);
-		if (StringUtils.isNotEmpty(from))    emailXml.addElement("from", from);
-		if (StringUtils.isNotEmpty(replyTo)) emailXml.addElement("replyTo", replyTo);
-		if (StringUtils.isNotEmpty(sender)) emailXml.addElement("sender", sender);
+		try (SaxElementBuilder recipientsXml = emailXml.startElement("recipients")) {
+			addRecipientProperty(recipientsXml, properties, IMailFileSystem.TO_RECEPIENTS_KEY);
+			addRecipientProperty(recipientsXml, properties, IMailFileSystem.CC_RECEPIENTS_KEY);
+			addRecipientProperty(recipientsXml, properties, IMailFileSystem.BCC_RECEPIENTS_KEY);
+		}
+		addPropertyAsHeader(emailXml,IMailFileSystem.FROM_ADDRESS_KEY, properties.get(IMailFileSystem.FROM_ADDRESS_KEY));
+		addPropertyAsHeader(emailXml,IMailFileSystem.SENDER_ADDRESS_KEY, properties.get(IMailFileSystem.SENDER_ADDRESS_KEY));
+		addPropertyAsHeader(emailXml,IMailFileSystem.REPLY_TO_RECEPIENTS_KEY, properties.get(IMailFileSystem.REPLY_TO_RECEPIENTS_KEY));
+		String bestReplyAddress = findBestReplyAddress(properties);
 		emailXml.addElement("bestReplyAddress", bestReplyAddress);
 		emailXml.addElement("subject", fileSystem.getSubject(emailMessage));
+		addPropertyAsHeader(emailXml,IMailFileSystem.DATETIME_SENT_KEY, properties.get(IMailFileSystem.DATETIME_SENT_KEY));
+		addPropertyAsHeader(emailXml,IMailFileSystem.DATETIME_RECEIVED_KEY, properties.get(IMailFileSystem.DATETIME_RECEIVED_KEY));
 		emailXml.addElement("message", fileSystem.getMessageBody(emailMessage));
 		try (SaxElementBuilder attachmentsXml = emailXml.startElement("attachments")) {
 			for (Iterator<A> it = fileSystem.listAttachments(emailMessage); it.hasNext();) {
@@ -72,19 +108,12 @@ public class MailFileSystemUtils {
 		try (SaxElementBuilder headersXml = emailXml.startElement("headers")) {
 			if (properties != null) {
 				for (Map.Entry<String,Object> header: properties.entrySet()) {
-					if (header.getValue() instanceof List) {
-						for(Object value:(List<?>)header.getValue()) {
-							headersXml.addElement("header", "name", header.getKey(), (String)value);
-						}
-					} else {
-						headersXml.addElement("header", "name", header.getKey(), (String)header.getValue());
+					if (!specialHeaders.contains(header.getKey())) {
+						addPropertyAsHeader(headersXml, "header", "name", header.getKey(), header.getValue());
 					}
 				}
 			}
 		}
-		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
-		emailXml.addElement("dateTimeSent", sdf.format(fileSystem.getDateTimeSent(emailMessage)));
-		emailXml.addElement("dateTimeReceived", sdf.format(fileSystem.getDateTimeReceived(emailMessage)));
 	}
 
 	public static <M,A> void addAttachmentInfo(IMailFileSystem<M,A> fileSystem, A attachment, SaxElementBuilder attachmentsXml) throws FileSystemException, SAXException {
@@ -102,23 +131,39 @@ public class MailFileSystemUtils {
 	}
 	
 
-	public static String findBestReplyAddress(String from, String replyTo, String sender, Map<String,Object> headers) {
+	public static String findBestReplyAddress(Map<String,Object> headers) {
 		String result;
 		
-		if (null != (result = getValidAddress("replyTo", replyTo))) {
+		if (null != (result = getValidAddressFromHeader(IMailFileSystem.REPLY_TO_RECEPIENTS_KEY, headers))) {
 			return result; 
 		}
-		if (null != (result = getValidAddress("from", from))) {
+		if (null != (result = getValidAddressFromHeader(IMailFileSystem.FROM_ADDRESS_KEY, headers))) {
 			return result; 
 		}
-		if (null != (result = getValidAddress("sender", sender))) {
+		if (null != (result = getValidAddressFromHeader(IMailFileSystem.SENDER_ADDRESS_KEY, headers))) {
 			return result; 
 		}
-		Object returnPath = headers.get(RETURN_PATH_HEADER);
-		if (returnPath instanceof String && null != (result = getValidAddress("return path", (String)returnPath))) {
+		if (null != (result = getValidAddressFromHeader(IMailFileSystem.RETURN_PATH_HEADER, headers))) {
 			return result; 
 		}
 		return null;
+	}
+
+	public static String getValidAddressFromHeader(String key, Map<String,Object> headers) {
+		Object item = headers.get(key);
+		if (item == null) {
+			return null;
+		}
+		if (item instanceof List) {
+			String result;
+			for(String address:(List<String>) item) {
+				if (null != (result = getValidAddress(key, address))) {
+					return result; 
+				}
+			}
+			return null;
+		}
+		return getValidAddress(key, item.toString());
 	}
 	
 	public static String getValidAddress(String type, String address) {

--- a/core/src/main/java/nl/nn/adapterframework/receivers/ExchangeMailListener.java
+++ b/core/src/main/java/nl/nn/adapterframework/receivers/ExchangeMailListener.java
@@ -94,16 +94,21 @@ public class ExchangeMailListener extends MailListener<EmailMessage,Attachment,E
 	}
 
 	@IbisDocRef({"9", EXCHANGE_FILE_SYSTEM})
+	public void setReplyAddressFields(String replyAddressFields) {
+		getFileSystem().setReplyAddressFields(replyAddressFields);
+	}
+	
+	@IbisDocRef({"10", EXCHANGE_FILE_SYSTEM})
 	public void setProxyHost(String proxyHost) {
 		getFileSystem().setProxyHost(proxyHost);
 	}
 
-	@IbisDocRef({"10", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"11", EXCHANGE_FILE_SYSTEM})
 	public void setProxyPort(int proxyPort) {
 		getFileSystem().setProxyPort(proxyPort);
 	}
 
-	@IbisDocRef({"11", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"12", EXCHANGE_FILE_SYSTEM})
 	public void setProxyUsername(String proxyUsername) {
 		getFileSystem().setProxyUsername(proxyUsername);
 	}
@@ -112,17 +117,17 @@ public class ExchangeMailListener extends MailListener<EmailMessage,Attachment,E
 	public void setProxyUserName(String proxyUsername) {
 		setProxyUsername(proxyUsername);
 	}
-	@IbisDocRef({"12", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"13", EXCHANGE_FILE_SYSTEM})
 	public void setProxyPassword(String proxyPassword) {
 		getFileSystem().setProxyPassword(proxyPassword);
 	}
 
-	@IbisDocRef({"13", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"14", EXCHANGE_FILE_SYSTEM})
 	public void setProxyAuthAlias(String proxyAuthAlias) {
 		getFileSystem().setProxyAuthAlias(proxyAuthAlias);
 	}
 
-	@IbisDocRef({"14", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"15", EXCHANGE_FILE_SYSTEM})
 	public void setProxyDomain(String domain) {
 		getFileSystem().setProxyDomain(domain);
 	}

--- a/core/src/main/java/nl/nn/adapterframework/senders/ExchangeFolderSender.java
+++ b/core/src/main/java/nl/nn/adapterframework/senders/ExchangeFolderSender.java
@@ -79,16 +79,21 @@ public class ExchangeFolderSender extends FileSystemSender<EmailMessage,Exchange
 	}
 
 	@IbisDocRef({"9", EXCHANGE_FILE_SYSTEM})
+	public void setReplyAddressFields(String replyAddressFields) {
+		getFileSystem().setReplyAddressFields(replyAddressFields);
+	}
+	
+	@IbisDocRef({"10", EXCHANGE_FILE_SYSTEM})
 	public void setProxyHost(String proxyHost) {
 		getFileSystem().setProxyHost(proxyHost);
 	}
 
-	@IbisDocRef({"10", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"11", EXCHANGE_FILE_SYSTEM})
 	public void setProxyPort(int proxyPort) {
 		getFileSystem().setProxyPort(proxyPort);
 	}
 
-	@IbisDocRef({"11", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"12", EXCHANGE_FILE_SYSTEM})
 	public void setProxyUsername(String proxyUsername) {
 		getFileSystem().setProxyUsername(proxyUsername);
 	}
@@ -97,17 +102,17 @@ public class ExchangeFolderSender extends FileSystemSender<EmailMessage,Exchange
 	public void setProxyUserName(String proxyUsername) {
 		setProxyUsername(proxyUsername);
 	}
-	@IbisDocRef({"12", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"13", EXCHANGE_FILE_SYSTEM})
 	public void setProxyPassword(String proxyPassword) {
 		getFileSystem().setProxyPassword(proxyPassword);
 	}
 
-	@IbisDocRef({"13", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"14", EXCHANGE_FILE_SYSTEM})
 	public void setProxyAuthAlias(String proxyAuthAlias) {
 		getFileSystem().setProxyAuthAlias(proxyAuthAlias);
 	}
 
-	@IbisDocRef({"14", EXCHANGE_FILE_SYSTEM})
+	@IbisDocRef({"15", EXCHANGE_FILE_SYSTEM})
 	public void setProxyDomain(String domain) {
 		getFileSystem().setProxyDomain(domain);
 	}

--- a/core/src/test/java/nl/nn/adapterframework/filesystem/MailFileSystemUtilsTest.java
+++ b/core/src/test/java/nl/nn/adapterframework/filesystem/MailFileSystemUtilsTest.java
@@ -1,0 +1,40 @@
+package nl.nn.adapterframework.filesystem;
+
+import static org.junit.Assert.assertEquals;
+
+import javax.mail.internet.AddressException;
+
+import org.junit.Test;
+
+public class MailFileSystemUtilsTest {
+
+	public void testGetValidAddress(String address) throws AddressException {
+		testGetValidAddress(address, address);
+	}
+	
+	public void testGetValidAddress(String expected, String address) throws AddressException {
+		assertEquals(expected, MailFileSystemUtils.getValidAddress("test", address));
+	}
+
+	@Test
+	public void testGetValidAddress() throws AddressException {
+		testGetValidAddress("xxx@yy.nl");
+		testGetValidAddress("Xxx <xxx@yy.nl>");
+		testGetValidAddress("Xxx <xxx@yy.nl>", "\"Xxx\" <xxx@yy.nl>");
+	}
+
+	@Test
+	public void testgetValidAddressWithWhitespace1() throws AddressException {
+		testGetValidAddress(null, "scan@ <popp.dk scan@popp.dk>");
+	}
+
+	@Test
+	public void testGetValidAddresssWithWhitespace2() throws AddressException {
+		testGetValidAddress(null, "abc xxx@yy.nl");
+	}
+
+	@Test
+	public void testGetValidAddressWithWhitespace3() throws AddressException {
+		testGetValidAddress(null, "xxx@yy.nl xxx@yy.nl");
+	}
+}

--- a/test-integration/src/test/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemTest.java
+++ b/test-integration/src/test/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemTest.java
@@ -104,6 +104,26 @@ public class ExchangeFileSystemTest extends SelfContainedBasicFileSystemTest<Ema
 	
 	
 	@Test
+	public void testExtractMessageWithProblematicAddress2() throws Exception {
+		EmailMessage emailMessage = getFirstFileFromFolder("FromAddressProblem2");
+		SaxElementBuilder xml = new SaxElementBuilder("email");
+		fileSystem.extractEmail(emailMessage, xml);
+		xml.close();
+		String expected = TestFileUtils.getTestFile("/ExchangeMailFromAddressProblem2.xml");
+		MatchUtils.assertXmlEquals(expected,xml.toString());
+	}
+	
+	@Test
+	public void testExtractMessageWithProblematicAddress3() throws Exception {
+		EmailMessage emailMessage = getFirstFileFromFolder("FromAddressProblem3");
+		SaxElementBuilder xml = new SaxElementBuilder("email");
+		fileSystem.extractEmail(emailMessage, xml);
+		xml.close();
+		String expected = TestFileUtils.getTestFile("/ExchangeMailFromAddressProblem3.xml");
+		MatchUtils.assertXmlEquals(expected,xml.toString());
+	}
+	
+	@Test
 	public void testExtractMessageWithMessageAttached() throws Exception {
 		EmailMessage emailMessage = getFirstFileFromFolder("AttachedMessage");
 		SaxElementBuilder xml = new SaxElementBuilder("email");

--- a/test-integration/src/test/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemTest.java
+++ b/test-integration/src/test/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemTest.java
@@ -91,6 +91,17 @@ public class ExchangeFileSystemTest extends SelfContainedBasicFileSystemTest<Ema
 		String expected = TestFileUtils.getTestFile("/ExchangeMailProblem.xml");
 		MatchUtils.assertXmlEquals(expected,xml.toString());
 	}
+
+	@Test
+	public void testExtractMessageWithProblematicAddress() throws Exception {
+		EmailMessage emailMessage = getFirstFileFromFolder("FromAddressProblem");
+		SaxElementBuilder xml = new SaxElementBuilder("email");
+		fileSystem.extractEmail(emailMessage, xml);
+		xml.close();
+		String expected = TestFileUtils.getTestFile("/ExchangeMailFromAddressProblem.xml");
+		MatchUtils.assertXmlEquals(expected,xml.toString());
+	}
+	
 	
 	@Test
 	public void testExtractMessageWithMessageAttached() throws Exception {

--- a/test-integration/src/test/resources/ExchangeMailAttachedMessage.xml
+++ b/test-integration/src/test/resources/ExchangeMailAttachedMessage.xml
@@ -3,8 +3,11 @@
 		<recipient type="to">gerrit@wearefrank.nl</recipient>
 	</recipients>
 	<from>gerrit@wearefrank.nl</from>
+	<sender>gerrit@wearefrank.nl</sender>
 	<bestReplyAddress>gerrit@wearefrank.nl</bestReplyAddress>
 	<subject>attached messages</subject>
+	<DateTimeSent>2020-11-10T11:01:48.000+0100</DateTimeSent>
+	<DateTimeReceived>2020-11-10T11:01:58.000+0100</DateTimeReceived>
 	<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;style type="text/css" style="display:none;"&gt; P {margin-top:0;margin-bottom:0;} &lt;/style&gt;&lt;/head&gt;&lt;body dir="ltr"&gt;&lt;div style="font-family: Calibri, Arial, Helvetica, sans-serif; font-size: 12pt; color: rgb(0, 0, 0);"&gt;probeer het maar!&lt;/div&gt;&lt;div&gt;&lt;div id="Signature"&gt;&lt;div&gt;&lt;div id="divtagdefaultwrapper" dir="ltr" style="font-size:12pt; color:#000000; font-family:Calibri,Helvetica,sans-serif"&gt;&lt;p style="margin-top: 0px; margin-bottom: 0px;"&gt;&lt;/p&gt;&lt;/div&gt;&lt;/div&gt;&lt;/div&gt;&lt;/div&gt;&lt;/body&gt;&lt;/html&gt;</message>
 	<attachments>
@@ -14,10 +17,12 @@
 				<recipient type="cc">subscribed@noreply.github.com</recipient>
 			</recipients>
 			<from>notifications@github.com</from>
-			<replyTo>nullreply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</replyTo>
 			<sender>notifications@github.com</sender>
-			<bestReplyAddress>nullreply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</bestReplyAddress>
+			<replyTo>reply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</replyTo>
+			<bestReplyAddress>reply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</bestReplyAddress>
 			<subject>Re: [ibissource/iaf] Ladybug randomly stops report generation (#875)</subject>
+			<DateTimeSent>2020-09-30T12:35:41.000+0200</DateTimeSent>
+			<DateTimeReceived>2020-09-30T12:35:45.000+0200</DateTimeReceived>
 			<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;/head&gt;&lt;body&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;Closed &lt;a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="644549868" data-permission-text="Title is private" data-url="https://github.com/ibissource/iaf/issues/875" data-hovercard-type="issue" data-hovercard-url="/ibissource/iaf/issues/875/hovercard" href="https://github.com/ibissource/iaf/issues/875"&gt;#875&lt;/a&gt;.&lt;/p&gt;&lt;p style="font-size:small;-webkit-text-size-adjust:none;color:#666;"&gt;—&lt;br&gt;You are receiving this because you are subscribed to this thread.&lt;br&gt;Reply to this email directly, &lt;a href="https://github.com/ibissource/iaf/issues/875#event-3823828368"&gt;view it on GitHub&lt;/a&gt;, or &lt;a href="https://github.com/notifications/unsubscribe-auth/AA72C6EQCEEQB2J2GMWU6JTSIMCX3ANCNFSM4OGSSZKA"&gt;unsubscribe&lt;/a&gt;.&lt;img src="https://github.com/notifications/beacon/AA72C6E5QJWXL4YUN7HSV2LSIMCX3A5CNFSM4OGSSZKKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGO4PVP3EA.gif" height="1" width="1" alt=""&gt;&lt;/p&gt;&lt;script type="application/ld+json"&gt;[
 {
@@ -95,8 +100,6 @@
 				<header name="X-Microsoft-Antispam-Mailbox-Delivery">ucf:1;jmr:0;auth:0;dest:C;OFR:CustomRules;ENG:(20160514016)(750128)(520011016)(944506458)(944626604);</header>
 				<header name="X-Microsoft-Antispam-Message-Info">/Z1KDPJKiwvjTAxcjszxs3Hv+PUemmrP+Y3Np4z1DWjGkYE37y0BFpvjCEp43XokULBpqMgsX6aiwNUtpQNiqW+0eahtAjnrWhgVGKLWDtEZRqcqz7Us+8qXvA+9HTRPjSP5UXpABiuVRssQP3YGksYSPfYkSyXAOMwtlqUBYNyvoa5Sxp4SYF3djYaWe4FUvcmwvm5X/PmZJvMx8vRhh53fMK82UWaxUc4I6lCb4ikhZg3ypUhtqul8PCjEM0iJf7nFN2+Ao8hyI/o0frg8atAkzqWD91wlYOt7frIoasxB9lALtfy9q17q/S6wo+EUUUbqRM5N+7Nth+1LGmZlqodnegnPl6d7fIsw7TrnjzrZVGeqWFUqh/uuq3CHD+cLdVEZRVQ+OciZuwisFB6dtH8HAzRTOFL9njjRyfvWUXG9OwtiVUX5VCcOA83mvfm2NJyXdXuA/FikPuElZ1fYHNE7+Eec0rs/t5G0qF/We68QELVVYSn5rVLk7kh4KCc7qO11JLmMgq/uEaz36LqgP1UAfya/8ytV97xXnmm8AZYiPNMe4wk+0AG+BhRb90S59tza/Ok/yDWNaoNTei7IyYVanlqweciI42kpJkU1rtb3XmqN4tKsBipmmY1SxRiYBPOiWjTal7gDPB/cV28UzgBLx4xcWPqOAyWWSdbXl31UK8+OHLZ5b0qujYzFydtfCbzpGd8/yfvtLv39s04T9vvuLFfp2aD35dogMNDUvloJsgHp7pIpwwDzzH/tFsyMoZLbLR4LgxSuuiSgyCVatUxPWnmAcK6KQXlhuxaBOTBgNxqxCJqYW+mTLc0nnYATXTmMm4S/QbjOaDnkMeko+4l5Xp9VYBO75TlNcry3mkfamT6D7oJpc+IDMgmqes/0N32M6eV4ssuB92a5jTt7NiBX8OYoMoP/hJKbKqU9AdRja8NfaI4+9AsxuoOSp7H09XkTFpp+Mnqxjx11H/u8WXsg2KekmPX/w0L9ku6+m6YzaVsKE4DOk0EnX84det/GEbyrpMn5gAQImHrwaNOqxzUtFgsWCzvKrLdhKnJn9YJRhSM3soRUb6sgAaqXWrT529U2mMrV8GGCTUvZITgTVMG2YZPPZvd8G03ZwlDZjThBaNJ5snZ2VBgIonqSeVx9W66Nt5lUW6tb7dnKUL7rFm1cGnTDl5lTM8cNUSCJ4jLX44eApUfVn/Mk9dequQy3msrSwQ3OZGOfVSzn6sanEtEkwDbYr5mHtGUOUjKzrDyxMtL7pflbqZhy3/gUR3QT4j5+qc1YtVThYDLdkg4HMtlqlw8iJbUtE2cjyySxlC9k3bp7aEpgkewvVRowqNt96it3MiDHEgncSouvCJqoGd4vPGlZ1u5XJz26R3ThOxvVmZwpfyaxR10WLe+pHs0Jx8oY/0Bxg8Y29gIcjBgCJM+jVLqUnSllBuYp7mYrloTS1zmCXY/n39VKlUav9ZoJGgNqng/g53JV8ZiCEZaBNAVnCDCPVcbg/an4La6R3s1nro3a3jQHDDXxMA1qsEC6</header>
 			</headers>
-			<dateTimeSent>2020-09-30T12:35:41.000+0200</dateTimeSent>
-			<dateTimeReceived>2020-09-30T12:35:45.000+0200</dateTimeReceived>
 		</attachment>
 	</attachments>
 	<headers>
@@ -149,6 +152,4 @@
 		<header name="X-Microsoft-Antispam-Mailbox-Delivery">ucf:0;jmr:0;auth:0;dest:I;ENG:(750128)(520011016)(706158)(944506458)(944626604);</header>
 		<header name="X-Microsoft-Antispam-Message-Info">4Cvnj240dSqSen2E82+130i0ICNgty9m1Gq7Ye6wTpoG4fHfpFlT+0c/AWtpnGWk/nxo5J5mJvyh8jZ5+adRb6GLgodC+SHdG+IH6bG3GNqgLMfXO3JG8feSmPsaVC0GJIIwDibhxGK4o8k9U1Cxr1bPV/AXsvkIbASc87SG72EKip9U2oXAVTCn95r94rJC/EXXxkwGlqA8N0aPp66AyIJXBkM0JCimHMt/VAXcJDbaKLsDMnZBEW2zzEkDMoOdADOZtGt5kt8d0AuALp//C7oYE0LHWvmGaaQCBwYkv5Ex9efyktQnulbsAdoY0kTZERgkNt/645wgUchkosDlBtRpIxUFPFmx/GXP7vVmgYfg7oXY21vFR4onxjjGYxQ7h1ILDicldsjTfY+RunQ6Qh4Ors5YgmiPaUm7MHkIH+E=</header>
 	</headers>
-	<dateTimeSent>2020-11-10T11:01:48.000+0100</dateTimeSent>
-	<dateTimeReceived>2020-11-10T11:01:58.000+0100</dateTimeReceived>
 </email>

--- a/test-integration/src/test/resources/ExchangeMailAttachedMessage.xml
+++ b/test-integration/src/test/resources/ExchangeMailAttachedMessage.xml
@@ -3,6 +3,7 @@
 		<recipient type="to">gerrit@wearefrank.nl</recipient>
 	</recipients>
 	<from>gerrit@wearefrank.nl</from>
+	<bestReplyAddress>gerrit@wearefrank.nl</bestReplyAddress>
 	<subject>attached messages</subject>
 	<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;style type="text/css" style="display:none;"&gt; P {margin-top:0;margin-bottom:0;} &lt;/style&gt;&lt;/head&gt;&lt;body dir="ltr"&gt;&lt;div style="font-family: Calibri, Arial, Helvetica, sans-serif; font-size: 12pt; color: rgb(0, 0, 0);"&gt;probeer het maar!&lt;/div&gt;&lt;div&gt;&lt;div id="Signature"&gt;&lt;div&gt;&lt;div id="divtagdefaultwrapper" dir="ltr" style="font-size:12pt; color:#000000; font-family:Calibri,Helvetica,sans-serif"&gt;&lt;p style="margin-top: 0px; margin-bottom: 0px;"&gt;&lt;/p&gt;&lt;/div&gt;&lt;/div&gt;&lt;/div&gt;&lt;/div&gt;&lt;/body&gt;&lt;/html&gt;</message>
@@ -13,6 +14,9 @@
 				<recipient type="cc">subscribed@noreply.github.com</recipient>
 			</recipients>
 			<from>notifications@github.com</from>
+			<replyTo>nullreply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</replyTo>
+			<sender>notifications@github.com</sender>
+			<bestReplyAddress>nullreply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</bestReplyAddress>
 			<subject>Re: [ibissource/iaf] Ladybug randomly stops report generation (#875)</subject>
 			<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;/head&gt;&lt;body&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;Closed &lt;a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="644549868" data-permission-text="Title is private" data-url="https://github.com/ibissource/iaf/issues/875" data-hovercard-type="issue" data-hovercard-url="/ibissource/iaf/issues/875/hovercard" href="https://github.com/ibissource/iaf/issues/875"&gt;#875&lt;/a&gt;.&lt;/p&gt;&lt;p style="font-size:small;-webkit-text-size-adjust:none;color:#666;"&gt;—&lt;br&gt;You are receiving this because you are subscribed to this thread.&lt;br&gt;Reply to this email directly, &lt;a href="https://github.com/ibissource/iaf/issues/875#event-3823828368"&gt;view it on GitHub&lt;/a&gt;, or &lt;a href="https://github.com/notifications/unsubscribe-auth/AA72C6EQCEEQB2J2GMWU6JTSIMCX3ANCNFSM4OGSSZKA"&gt;unsubscribe&lt;/a&gt;.&lt;img src="https://github.com/notifications/beacon/AA72C6E5QJWXL4YUN7HSV2LSIMCX3A5CNFSM4OGSSZKKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGO4PVP3EA.gif" height="1" width="1" alt=""&gt;&lt;/p&gt;&lt;script type="application/ld+json"&gt;[

--- a/test-integration/src/test/resources/ExchangeMailNormal.xml
+++ b/test-integration/src/test/resources/ExchangeMailNormal.xml
@@ -4,6 +4,7 @@
 		<recipient type="cc">subscribed@noreply.github.com</recipient>
 	</recipients>
 	<from>notifications@github.com</from>
+	<bestReplyAddress>notifications@github.com</bestReplyAddress>
 	<subject>Re: [ibissource/iaf] Ladybug randomly stops report generation (#875)</subject>
 	<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;/head&gt;&lt;body&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;Closed &lt;a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="644549868" data-permission-text="Title is private" data-url="https://github.com/ibissource/iaf/issues/875" data-hovercard-type="issue" data-hovercard-url="/ibissource/iaf/issues/875/hovercard" href="https://github.com/ibissource/iaf/issues/875"&gt;#875&lt;/a&gt;.&lt;/p&gt;&lt;p style="font-size:small;-webkit-text-size-adjust:none;color:#666;"&gt;—&lt;br&gt;You are receiving this because you are subscribed to this thread.&lt;br&gt;Reply to this email directly, &lt;a href="https://github.com/ibissource/iaf/issues/875#event-3823828368"&gt;view it on GitHub&lt;/a&gt;, or &lt;a href="https://github.com/notifications/unsubscribe-auth/AA72C6EQCEEQB2J2GMWU6JTSIMCX3ANCNFSM4OGSSZKA"&gt;unsubscribe&lt;/a&gt;.&lt;img src="https://github.com/notifications/beacon/AA72C6E5QJWXL4YUN7HSV2LSIMCX3A5CNFSM4OGSSZKKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGO4PVP3EA.gif" height="1" width="1" alt=""&gt;&lt;/p&gt;&lt;script type="application/ld+json"&gt;[

--- a/test-integration/src/test/resources/ExchangeMailNormal.xml
+++ b/test-integration/src/test/resources/ExchangeMailNormal.xml
@@ -4,8 +4,12 @@
 		<recipient type="cc">subscribed@noreply.github.com</recipient>
 	</recipients>
 	<from>notifications@github.com</from>
-	<bestReplyAddress>notifications@github.com</bestReplyAddress>
+	<sender>notifications@github.com</sender>
+	<replyTo>reply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</replyTo>
+	<bestReplyAddress>reply+AA72C6GJP3QDZ2ZAJ4NOTZN5QBAH3EVBNHHCM2YM5Q@reply.github.com</bestReplyAddress>
 	<subject>Re: [ibissource/iaf] Ladybug randomly stops report generation (#875)</subject>
+	<DateTimeSent>2020-09-30T12:35:41.000+0200</DateTimeSent>
+	<DateTimeReceived>2020-09-30T12:35:45.000+0200</DateTimeReceived>
 	<message>&lt;html&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;/head&gt;&lt;body&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;Closed &lt;a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="644549868" data-permission-text="Title is private" data-url="https://github.com/ibissource/iaf/issues/875" data-hovercard-type="issue" data-hovercard-url="/ibissource/iaf/issues/875/hovercard" href="https://github.com/ibissource/iaf/issues/875"&gt;#875&lt;/a&gt;.&lt;/p&gt;&lt;p style="font-size:small;-webkit-text-size-adjust:none;color:#666;"&gt;—&lt;br&gt;You are receiving this because you are subscribed to this thread.&lt;br&gt;Reply to this email directly, &lt;a href="https://github.com/ibissource/iaf/issues/875#event-3823828368"&gt;view it on GitHub&lt;/a&gt;, or &lt;a href="https://github.com/notifications/unsubscribe-auth/AA72C6EQCEEQB2J2GMWU6JTSIMCX3ANCNFSM4OGSSZKA"&gt;unsubscribe&lt;/a&gt;.&lt;img src="https://github.com/notifications/beacon/AA72C6E5QJWXL4YUN7HSV2LSIMCX3A5CNFSM4OGSSZKKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGO4PVP3EA.gif" height="1" width="1" alt=""&gt;&lt;/p&gt;&lt;script type="application/ld+json"&gt;[
 {
@@ -83,6 +87,4 @@
 		<header name="X-Microsoft-Antispam-Mailbox-Delivery">ucf:1;jmr:0;auth:0;dest:C;OFR:CustomRules;ENG:(20160514016)(750128)(520011016)(944506458)(944626604);</header>
 		<header name="X-Microsoft-Antispam-Message-Info">/Z1KDPJKiwvjTAxcjszxs3Hv+PUemmrP+Y3Np4z1DWjGkYE37y0BFpvjCEp43XokULBpqMgsX6aiwNUtpQNiqW+0eahtAjnrWhgVGKLWDtEZRqcqz7Us+8qXvA+9HTRPjSP5UXpABiuVRssQP3YGksYSPfYkSyXAOMwtlqUBYNyvoa5Sxp4SYF3djYaWe4FUvcmwvm5X/PmZJvMx8vRhh53fMK82UWaxUc4I6lCb4ikhZg3ypUhtqul8PCjEM0iJf7nFN2+Ao8hyI/o0frg8atAkzqWD91wlYOt7frIoasxB9lALtfy9q17q/S6wo+EUUUbqRM5N+7Nth+1LGmZlqodnegnPl6d7fIsw7TrnjzrZVGeqWFUqh/uuq3CHD+cLdVEZRVQ+OciZuwisFB6dtH8HAzRTOFL9njjRyfvWUXG9OwtiVUX5VCcOA83mvfm2NJyXdXuA/FikPuElZ1fYHNE7+Eec0rs/t5G0qF/We68QELVVYSn5rVLk7kh4KCc7qO11JLmMgq/uEaz36LqgP1UAfya/8ytV97xXnmm8AZYiPNMe4wk+0AG+BhRb90S59tza/Ok/yDWNaoNTei7IyYVanlqweciI42kpJkU1rtb3XmqN4tKsBipmmY1SxRiYBPOiWjTal7gDPB/cV28UzgBLx4xcWPqOAyWWSdbXl31UK8+OHLZ5b0qujYzFydtfCbzpGd8/yfvtLv39s04T9vvuLFfp2aD35dogMNDUvloJsgHp7pIpwwDzzH/tFsyMoZLbLR4LgxSuuiSgyCVatUxPWnmAcK6KQXlhuxaBOTBgNxqxCJqYW+mTLc0nnYATXTmMm4S/QbjOaDnkMeko+4l5Xp9VYBO75TlNcry3mkfamT6D7oJpc+IDMgmqes/0N32M6eV4ssuB92a5jTt7NiBX8OYoMoP/hJKbKqU9AdRja8NfaI4+9AsxuoOSp7H09XkTFpp+Mnqxjx11H/u8WXsg2KekmPX/w0L9ku6+m6YzaVsKE4DOk0EnX84det/GEbyrpMn5gAQImHrwaNOqxzUtFgsWCzvKrLdhKnJn9YJRhSM3soRUb6sgAaqXWrT529U2mMrV8GGCTUvZITgTVMG2YZPPZvd8G03ZwlDZjThBaNJ5snZ2VBgIonqSeVx9W66Nt5lUW6tb7dnKUL7rFm1cGnTDl5lTM8cNUSCJ4jLX44eApUfVn/Mk9dequQy3msrSwQ3OZGOfVSzn6sanEtEkwDbYr5mHtGUOUjKzrDyxMtL7pflbqZhy3/gUR3QT4j5+qc1YtVThYDLdkg4HMtlqlw8iJbUtE2cjyySxlC9k3bp7aEpgkewvVRowqNt96it3MiDHEgncSouvCJqoGd4vPGlZ1u5XJz26R3ThOxvVmZwpfyaxR10WLe+pHs0Jx8oY/0Bxg8Y29gIcjBgCJM+jVLqUnSllBuYp7mYrloTS1zmCXY/n39VKlUav9ZoJGgNqng/g53JV8ZiCEZaBNAVnCDCPVcbg/an4La6R3s1nro3a3jQHDDXxMA1qsEC6</header>
 	</headers>
-	<dateTimeSent>2020-09-30T12:35:41.000+0200</dateTimeSent>
-	<dateTimeReceived>2020-09-30T12:35:45.000+0200</dateTimeReceived>
 </email>

--- a/test-integration/src/test/resources/ExchangeMailProblem.xml
+++ b/test-integration/src/test/resources/ExchangeMailProblem.xml
@@ -3,6 +3,7 @@
 		<recipient type="to">gerrit@wearefrank.nl</recipient>
 	</recipients>
 	<from>Gerrit.van.Brakel@nn-group.com</from>
+	<bestReplyAddress>Gerrit.van.Brakel@nn-group.com</bestReplyAddress>
 	<subject>XML Problem #x3</subject>
 	<message>&lt;html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:w="urn:schemas-microsoft-com:office:word" xmlns:m="http://schemas.microsoft.com/office/2004/12/omml" xmlns="http://www.w3.org/TR/REC-html40"&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;meta name="Generator" content="Microsoft Word 15 (filtered medium)"&gt;&lt;!--[if !mso]&gt;&lt;style&gt;v\:* {behavior:url(#default#VML);}

--- a/test-integration/src/test/resources/ExchangeMailProblem.xml
+++ b/test-integration/src/test/resources/ExchangeMailProblem.xml
@@ -3,8 +3,11 @@
 		<recipient type="to">gerrit@wearefrank.nl</recipient>
 	</recipients>
 	<from>Gerrit.van.Brakel@nn-group.com</from>
+	<sender>Gerrit.van.Brakel@nn-group.com</sender>
 	<bestReplyAddress>Gerrit.van.Brakel@nn-group.com</bestReplyAddress>
 	<subject>XML Problem #x3</subject>
+	<DateTimeSent>2020-10-30T13:10:21.000+0100</DateTimeSent>
+	<DateTimeReceived>2020-10-30T13:10:29.000+0100</DateTimeReceived>
 	<message>&lt;html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:w="urn:schemas-microsoft-com:office:word" xmlns:m="http://schemas.microsoft.com/office/2004/12/omml" xmlns="http://www.w3.org/TR/REC-html40"&gt;&lt;head&gt;
 &lt;meta http-equiv="Content-Type" content="text/html; charset=utf-8"&gt;&lt;meta name="Generator" content="Microsoft Word 15 (filtered medium)"&gt;&lt;!--[if !mso]&gt;&lt;style&gt;v\:* {behavior:url(#default#VML);}
 o\:* {behavior:url(#default#VML);}
@@ -119,6 +122,4 @@ div.WordSection1
 		<header name="X-Microsoft-Antispam-Mailbox-Delivery">ucf:0;jmr:0;auth:0;dest:I;ENG:(20160514016)(750128)(520011016)(944506458)(944626604);</header>
 		<header name="X-Microsoft-Antispam-Message-Info">CjMgFkn3FRwDaixbneeCVaDzbbQeKQ5oBVdk4dQmQvgumynPCf6YJjEPU4yXicb0LlayizOaRpdoOv8/94UuXs30CLU75t2/aVEO4AilWKhY/v9w5Gz705MH8HlZiJmegOMh0RY3QheD6h9NYPSm7pWrYk4dV5g31wDq/FPtC1JHTSLIi0wuYu+8MXvPaUOryDHlM/5PTW6TblKNsSbA69AyhfS/50grpPxlLadyRENdnTfxSiwrMC/AiR0PlzUrn/+JVOQgHj/yObAQTO9W15Xu9SDOZs8uoCYTvmmG+HlWUJvWP+HtNSppZIK2DCNWN4P8lyedNiYezvfpK1UJREJDiB/Vjp4yZ0tcYKYGG3Bk5BTdTjoRNsktxdlANSRzmBs7xZi9M/VuiFq7bPuinQeiV+BA6d3w2jPYRs7F1zkxwLN2rcIoMTL4uNnrNwRq8w57WK0uFlIFt2ZfPtSprwCK0Jj2LnJVMHI9S8n8ZCIT6PkTJK9XwIUWN1dYsDPk/NJbV5ntHDSjJmSd6RMSlhthO4vBMxhlDT7XTQLZFnsCH1Bf5nDbNePd4RbW8x5Iaz/LBU1YOotjAdeqmLh0litxnQNAVaOdfctvWh26mxlqlNb4kB9C5nK/rmPOUiBGbd14W0xM9oqiIeqggp8c9R4cIOfidd8P/s/34SfsJtjFl1jEIL9qZnQycgqPoqSW5JBXkwLfP+bYD0Wub3Z3e8JHiFIvLk4auDVfaGsm2KptjcNJUxcHAchAUXzmhbthETaZ3FDB0sHdmgRiafNU3sCQUGQ0bvhrgPLlu1W7czro0i0YswzyoFDYYR3Ij8nIK5B1EGItbY9ni9Eo0obBuNUJMCpJYEJEEd1N0dmOY+HY4TRxDiz2kpsjBZfyxmwCTLhlxBF7iJ3RQxvjGXXN7O0b0aqL4KcC3UWaGoTLZJu6nwQBQR9UwjVZVxRUaIPX2hVJmkbGtQ9wgKRJ6EjfBRQRbhvfZm7TlDRMB1plYy9Ae58FDNGOuC5Q9N/LmL8IkKQgbw0o0cXmKt254CLjdwxwjFlBekASevA8XsxrclQ=</header>
 	</headers>
-	<dateTimeSent>2020-10-30T13:10:21.000+0100</dateTimeSent>
-	<dateTimeReceived>2020-10-30T13:10:29.000+0100</dateTimeReceived>
 </email>

